packages/client/components/LoadingSpinner.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { View, Animated, Easing } from "react-native";
+
+const LoadingSpinner = () => {
+    const spinValue = new Animated.Value(0);
+
+    Animated.loop(
+        Animated.timing(spinValue, {
+            toValue: 1,
+            duration: 2000,
+            useNativeDriver: true,
+            easing: Easing.linear,
+        }),
+    ).start();
+
+    const spin = spinValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: ["0deg", "360deg"],
+    });
+
+    return (
+        <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+            <Animated.Image
+                style={{ transform: [{ rotate: spin }] }}
+                source={require("../assets/images/Spinner.png")}
+            />
+        </View>
```

Comment:
I wouldnt wrap the component in a view or any type of layout styling so you can leave it up to the client, the component that uses this, to decide how to position it.

I think for how we use it now the positioning should already be done, but double check and add any new layout stuff to where its used.

```suggestion
        <Animated.Image
            style={{ transform: [{ rotate: spin }] }}
            source={require("../assets/images/Spinner.png")}
        />
```